### PR TITLE
Compute step and remaining duration from map.

### PR DESCRIPTION
**Pull request by afck**

### CODE REVIEW

File: ethcore/src/engines/authority_round/mod.rs [hunk 1]
```diff
@@ -1302,8 +1300,14 @@ impl Engine<EthereumMachine> for AuthorityRound {
...
 
+		let opt_signer = self.signer.read();
+		let signer = match opt_signer.as_ref() {
+			Some(signer) => signer,
+			None => return Ok(Vec::new()), // We are not a validator, so we shouldn't call the contracts.
+		};
+
 		// Our current account nonce. The transactions must have consecutive nonces, starting with this one.
-		let mut tx_nonce = block.state.nonce(&our_addr)?;
+		let mut tx_nonce = block.state.nonce(&signer.address())?;
```
Comment (vkomenda) [inline ethcore/src/engines/authority_round/mod.rs]: Why do you think it is better to call `address()` twice instead of using the reference `our_addr`?

Comment (afck) [reply ethcore/src/engines/authority_round/mod.rs]: Sorry, that was a mistake. I removed this part of the change.

**Comment by afck** [comment]
The [rebased version of this](https://github.com/poanetwork/parity-ethereum/tree/afck-step-duration-transition-rebased) passes the [posdao-test-setup](https://github.com/poanetwork/posdao-test-setup) tests for me.

**Comment by varasev** [comment]
Done in https://github.com/poanetwork/parity-ethereum/pull/152.

**Pull request closed without merging**